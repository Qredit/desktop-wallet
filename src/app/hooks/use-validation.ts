import { useMemo } from "react";
import { useTranslation } from "react-i18next";

import { useEnvironmentContext } from "@/app/contexts";
import { password } from "@/app/validations/password";
import { exchangeOrder } from "@/domains/exchange/validations";
import { createProfile } from "@/domains/profile/validations";
import { settings } from "@/domains/setting/validations";
import {
	authentication,
	common,
	delegateRegistration,
	multiSignatureRegistration,
	sendIpfs,
	sendTransfer,
	sendVote,
} from "@/domains/transaction/validations";
import { receiveFunds, verifyMessage } from "@/domains/wallet/validations";

export const useValidation = () => {
	const { t } = useTranslation();
	const { env } = useEnvironmentContext();

	return useMemo(
		() => ({
			authentication: authentication(t),
			common: common(t),
			createProfile: createProfile(t, env),
			delegateRegistration: delegateRegistration(t),
			exchangeOrder: exchangeOrder(t),
			multiSignatureRegistration: multiSignatureRegistration(t),
			password: password(t),
			receiveFunds: receiveFunds(t),
			sendIpfs: sendIpfs(t),
			sendTransfer: sendTransfer(t),
			sendVote: sendVote(t),
			settings: settings(t, env),
			verifyMessage: verifyMessage(t),
		}),
		[t, env],
	);
};
