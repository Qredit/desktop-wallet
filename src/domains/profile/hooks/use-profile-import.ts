import { Contracts, Environment } from "@payvo/sdk-profiles";

import { ReadableFile } from "@/app/hooks/use-files";

interface ImportFileProperties {
	file?: ReadableFile;
	password?: string;
}

export const useProfileImport = ({ env }: { env: Environment }) => {
	const importProfileFromDwe = async (profileData: string, password?: string) => {
		let profile: Contracts.IProfile;

		try {
			profile = await env.profiles().import(profileData, password);
		} catch (error) {
			if (error.message.includes("Reason: Unexpected token") && !password) {
				throw new Error("PasswordRequired");
			}

			if (error.message.includes("Reason: Unexpected token") && password) {
				throw new Error("InvalidPassword");
			}

			throw error;
		}

		return profile;
	};

	const importLegacyProfile = async (profileData: string) => {
		let data: Record<string, any>;

		try {
			data = JSON.parse(profileData);
		} catch {
			throw new Error("CorruptedData");
		}

		if (!data?.wallets?.length) {
			throw new Error("MissingWallets");
		}

		const profile = env.profiles().create("");

		for (const wallet of data.wallets) {
			if (wallet?.address && wallet?.balance.XQR) {
				await profile.coins().set("XQR", "xqr.mainnet").__construct();
			}

			if (wallet?.address && wallet?.balance.dXQR) {
				await profile.coins().set("XQR", "xqr.testnet").__construct();
			}
		}

		await Promise.all(
			data.wallets.map(async (wallet: Record<string, any>) => {
				if (wallet?.address && wallet?.balance.XQR) {
					const importedWallet = await profile.walletFactory().fromAddress({
						address: wallet.address,
						coin: "XQR",
						network: "xqr.mainnet",
					});
					profile.wallets().push(importedWallet);
					return wallet;
				}

				if (wallet?.address && wallet?.balance.dXQR) {
					profile.settings().set(Contracts.ProfileSetting.UseTestNetworks, true);

					const importedWallet = await profile.walletFactory().fromAddress({
						address: wallet.address,
						coin: "XQR",
						network: "xqr.testnet",
					});
					profile.wallets().push(importedWallet);
					return importedWallet;
				}
			}),
		);

		env.profiles().forget(profile.id());
		return profile;
	};

	const importProfile = async ({ password, file }: ImportFileProperties) => {
		if (!file) {
			return;
		}

		if (file?.extension === ".dwe") {
			return await importProfileFromDwe(file.content.toString(), password);
		}

		if (file?.extension === ".json") {
			return await importLegacyProfile(file.content.toString());
		}
	};

	return { importProfile };
};
