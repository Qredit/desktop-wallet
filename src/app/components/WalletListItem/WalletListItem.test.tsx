import { Contracts } from "@payvo/sdk-profiles";
import userEvent from "@testing-library/user-event";
import { createMemoryHistory } from "history";
import React from "react";
import { Route } from "react-router-dom";
import { WalletListItem } from "./WalletListItem";
import * as useWalletActionsModule from "@/domains/wallet/hooks/use-wallet-actions";

import { env, getDefaultProfileId, render, screen } from "@/utils/testing-library";
import * as isFullySyncedModule from "@/domains/wallet/utils/is-fully-synced";

const dashboardURL = `/profiles/${getDefaultProfileId()}/dashboard`;
const history = createMemoryHistory();

describe("WalletListItem", () => {
	let profile: Contracts.IProfile;
	let wallet: Contracts.IReadWriteWallet;

	beforeAll(() => {
		history.push(dashboardURL);
	});

	beforeEach(async () => {
		profile = env.profiles().findById(getDefaultProfileId());

		wallet = profile.wallets().findById("ac38fe6d-4b67-4ef1-85be-17c5f6841129");
		wallet.data().set(Contracts.WalletFlag.Starred, true);
		wallet.data().set(Contracts.WalletData.DerivationPath, "0");

		await env.profiles().restore(profile);
		await profile.sync();
	});

	it.each([true, false])("should render when isCompact = %s", (isCompact: boolean) => {
		const { container } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} isCompact={isCompact} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(screen.getByText(wallet.alias()!)).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it("should render with a N/A for fiat", () => {
		const isTestMock = jest.spyOn(wallet.network(), "isTest").mockReturnValue(true);

		const { asFragment } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} isCompact={false} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(asFragment()).toMatchSnapshot();
		expect(screen.getByText("N/A")).toBeInTheDocument();

		isTestMock.mockRestore();
	});

	it("should render with default BTC as default exchangeCurrency", () => {
		const mockExchangeCurrency = jest.spyOn(wallet, "exchangeCurrency").mockReturnValue(undefined as any);

		const { container } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} isCompact={false} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(screen.getByText(wallet.alias()!)).toBeInTheDocument();

		expect(container).toMatchSnapshot();

		mockExchangeCurrency.mockRestore();
	});

	it("should avoid click on TableRow when syncing", () => {
		const useWalletActionsReturn = {
			activeModal: "wallet-name",
			handleOpen: jest.fn(),
		} as unknown as ReturnType<typeof useWalletActionsModule.useWalletActions>;

		const isFullySyncedSpy = jest.spyOn(isFullySyncedModule, "isFullySynced").mockReturnValue(false);
		const useWalletActionsSpy = jest
			.spyOn(useWalletActionsModule, "useWalletActions")
			.mockReturnValue(useWalletActionsReturn);

		const { asFragment } = render(
			<table>
				<tbody>
					<Route path="/profiles/:profileId/dashboard">
						<WalletListItem wallet={wallet} isCompact={false} />
					</Route>
				</tbody>
			</table>,
			{
				history,
				routes: [dashboardURL],
			},
		);

		expect(asFragment).toMatchSnapshot();

		expect(screen.getByTestId("UpdateWalletName__submit")).toBeInTheDocument();

		userEvent.click(screen.getByTestId("TableRow"));

		expect(useWalletActionsReturn.handleOpen).not.toHaveBeenCalled();

		isFullySyncedSpy.mockRestore();
		useWalletActionsSpy.mockRestore();
	});
});
