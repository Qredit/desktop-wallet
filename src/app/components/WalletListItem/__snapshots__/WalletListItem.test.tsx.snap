// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletListItem should avoid click on TableRow when syncing 1`] = `[Function]`;

exports[`WalletListItem should render when isCompact = false 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 ijbCHm group"
        data-testid="TableRow"
      >
        <td
          data-testid="TableCell_Wallet"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 kjAxRM space-x-4"
          >
            <div
              class="flex items-center justify-center"
              data-testid="WalletIcon__Starred"
            >
              <div
                class="sc-bdvvtL jjGTFI text-theme-warning-400"
                height="18"
                width="18"
              >
                <svg>
                  star-filled.svg
                </svg>
              </div>
            </div>
            <div
              class="flex-shrink-0 flex items-center border-l border-theme-secondary-300 dark:border-theme-secondary-800 pl-4 ml-4"
            >
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 QrTVC ring-theme-background group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="flex-1 inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td
          data-testid="CurrencyCell"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cxSvRM justify-end text-theme-secondary-text"
          >
            <div
              data-testid="WalletHeader__send-button"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 cxgwmk my-auto"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  Send
                </div>
              </button>
            </div>
            <div
              class="ml-3"
              data-testid="WalletHeader__more-button"
            >
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <button
                    class="OriginalButton__StyledButton-sc-o0fs2j-0 cxgwmk w-11 flex-1 text-white bg-theme-primary-600 hover:bg-theme-primary-700"
                    type="button"
                  >
                    <div
                      class="flex items-center space-x-2"
                    >
                      <div
                        class="sc-bdvvtL qGXhc"
                        height="20"
                        width="20"
                      >
                        <svg>
                          ellipsis-vertical.svg
                        </svg>
                      </div>
                    </div>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WalletListItem should render when isCompact = true 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 ijbCHm group"
        data-testid="TableRow"
      >
        <td
          data-testid="TableCell_Wallet"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 eNekfl space-x-3"
          >
            <div
              class="flex items-center justify-center"
              data-testid="WalletIcon__Starred"
            >
              <div
                class="sc-bdvvtL jjGTFI text-theme-warning-400"
                height="18"
                width="18"
              >
                <svg>
                  star-filled.svg
                </svg>
              </div>
            </div>
            <div
              class="flex-shrink-0 flex items-center border-l border-theme-secondary-300 dark:border-theme-secondary-800 pl-3 ml-3"
            >
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 itzeiy ring-theme-background group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="flex-1 inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iTMWpD justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iTMWpD font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td
          data-testid="CurrencyCell"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iTMWpD justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 bfFMyp justify-end text-theme-secondary-text"
          >
            <div
              data-testid="WalletHeader__send-button"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF text-theme-primary-600 hover:text-theme-primary-700"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  Send
                </div>
              </button>
            </div>
            <div
              class=""
              data-testid="WalletHeader__more-button"
            >
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <button
                    class="OriginalButton__StyledButton-sc-o0fs2j-0 enzTPF w-11 -mr-1.5 text-theme-primary-300 hover:text-theme-primary-600"
                    type="button"
                  >
                    <div
                      class="flex items-center space-x-2"
                    >
                      <div
                        class="sc-bdvvtL qGXhc"
                        height="20"
                        width="20"
                      >
                        <svg>
                          ellipsis-vertical.svg
                        </svg>
                      </div>
                    </div>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`WalletListItem should render with a N/A for fiat 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 ijbCHm group"
        data-testid="TableRow"
      >
        <td
          data-testid="TableCell_Wallet"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 kjAxRM space-x-4"
          >
            <div
              class="flex items-center justify-center"
              data-testid="WalletIcon__Starred"
            >
              <div
                class="sc-bdvvtL jjGTFI text-theme-warning-400"
                height="18"
                width="18"
              >
                <svg>
                  star-filled.svg
                </svg>
              </div>
            </div>
            <div
              class="flex-shrink-0 flex items-center border-l border-theme-secondary-300 dark:border-theme-secondary-800 pl-4 ml-4"
            >
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 QrTVC ring-theme-background group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="flex-1 inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td
          data-testid="CurrencyCell"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cxSvRM justify-end text-theme-secondary-text"
          >
            <div
              data-testid="WalletHeader__send-button"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 cxgwmk my-auto"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  Send
                </div>
              </button>
            </div>
            <div
              class="ml-3"
              data-testid="WalletHeader__more-button"
            >
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <button
                    class="OriginalButton__StyledButton-sc-o0fs2j-0 cxgwmk w-11 flex-1 text-white bg-theme-primary-600 hover:bg-theme-primary-700"
                    type="button"
                  >
                    <div
                      class="flex items-center space-x-2"
                    >
                      <div
                        class="sc-bdvvtL qGXhc"
                        height="20"
                        width="20"
                      >
                        <svg>
                          ellipsis-vertical.svg
                        </svg>
                      </div>
                    </div>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`WalletListItem should render with default BTC as default exchangeCurrency 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-y9j4av-0 ijbCHm group"
        data-testid="TableRow"
      >
        <td
          data-testid="TableCell_Wallet"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 kjAxRM space-x-4"
          >
            <div
              class="flex items-center justify-center"
              data-testid="WalletIcon__Starred"
            >
              <div
                class="sc-bdvvtL jjGTFI text-theme-warning-400"
                height="18"
                width="18"
              >
                <svg>
                  star-filled.svg
                </svg>
              </div>
            </div>
            <div
              class="flex-shrink-0 flex items-center border-l border-theme-secondary-300 dark:border-theme-secondary-800 pl-4 ml-4"
            >
              <div
                class="Avatar__AvatarWrapper-sc-18vw8a4-0 QrTVC ring-theme-background group-hover:ring-theme-secondary-100 group-hover:bg-theme-secondary-100 dark:group-hover:ring-black dark:group-hover:bg-black flex-shrink-0"
                data-testid="Avatar"
              >
                <div
                  class="w-full h-full inline-flex items-center justify-center overflow-hidden align-middle rounded-full"
                >
                  <img
                    alt="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                    src="data:image/svg+xml;utf8,<svg version=\\"1.1\\" xmlns=\\"http://www.w3.org/2000/svg\\" class=\\"picasso\\" width=\\"100\\" height=\\"100\\" viewBox=\\"0 0 100 100\\"><style>.picasso circle{mix-blend-mode:soft-light;}</style><rect fill=\\"rgb(0, 150, 136)\\" width=\\"100\\" height=\\"100\\"/><circle r=\\"50\\" cx=\\"80\\" cy=\\"50\\" fill=\\"rgb(76, 175, 80)\\"/><circle r=\\"35\\" cx=\\"100\\" cy=\\"40\\" fill=\\"rgb(156, 39, 176)\\"/><circle r=\\"40\\" cx=\\"90\\" cy=\\"70\\" fill=\\"rgb(3, 169, 244)\\"/></svg>"
                    title="D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD"
                  />
                </div>
              </div>
            </div>
            <div
              class="w-20 flex-1"
            >
              <div
                class="flex overflow-hidden flex-grow items-center space-x-2 whitespace-nowrap no-ligatures"
              >
                <span
                  class="font-semibold text-base text-theme-text"
                  data-testid="Address__alias"
                >
                  <span
                    class="TruncateEnd__Wrapper-sc-1hhdnni-0"
                    data-testid="TruncateEnd"
                  >
                    ARK Wallet 1
                  </span>
                </span>
                <span
                  class="flex-1 inline-flex overflow-hidden text-theme-secondary-500 dark:text-theme-secondary-700 font-semibold text-base"
                  data-testid="Address__address"
                >
                  D8rr7B1d6TL6pf14LgMz4sKp1VBMs6YUYD
                </span>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq justify-center text-sm font-bold text-center align-middle"
          >
            <div
              class="inline-flex items-center space-x-1"
            >
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__Ledger"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    ledger.svg
                  </svg>
                </div>
              </div>
              <div
                class="inline-block p-1 text-theme-secondary-700"
                data-testid="WalletIcon__TestNetwork"
              >
                <div
                  class="sc-bdvvtL qGXhc"
                  height="20"
                  width="20"
                >
                  <svg>
                    code.svg
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq font-semibold justify-end"
          >
            <span
              class="whitespace-nowrap"
              data-testid="Amount"
            >
              33.75089801 DARK
            </span>
          </div>
        </td>
        <td
          data-testid="CurrencyCell"
        >
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 iWCFjq justify-end text-theme-secondary-text"
          >
            N/A
          </div>
        </td>
        <td>
          <div
            class="TableCell__TableCellInnerWrapper-sc-v39gua-0 cxSvRM justify-end text-theme-secondary-text"
          >
            <div
              data-testid="WalletHeader__send-button"
            >
              <button
                class="OriginalButton__StyledButton-sc-o0fs2j-0 cxgwmk my-auto"
                type="button"
              >
                <div
                  class="flex items-center space-x-2"
                >
                  Send
                </div>
              </button>
            </div>
            <div
              class="ml-3"
              data-testid="WalletHeader__more-button"
            >
              <div
                class="relative"
              >
                <span
                  data-testid="dropdown__toggle"
                >
                  <button
                    class="OriginalButton__StyledButton-sc-o0fs2j-0 cxgwmk w-11 flex-1 text-white bg-theme-primary-600 hover:bg-theme-primary-700"
                    type="button"
                  >
                    <div
                      class="flex items-center space-x-2"
                    >
                      <div
                        class="sc-bdvvtL qGXhc"
                        height="20"
                        width="20"
                      >
                        <svg>
                          ellipsis-vertical.svg
                        </svg>
                      </div>
                    </div>
                  </button>
                </span>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
